src/lib.rs

```diff
@@ -173,6 +173,42 @@ impl<'a> Sealer<'a> {
             },
         })
     }
+
+    pub fn new_from_layer(
+        provided_layer_index: usize,
+        provided_layer: Layer,
+        config: Config,
+        replica_id: Sha256Domain,
+        window_index: usize,
+        original_data: Layer,
+        gpu: &'a mut GPU,
+        build_trees: bool,
+        rows_to_discard: usize,
+    ) -> NSEResult<Self> {
```

Comment:
Too many parameters :astonished:

Reply:
How about we add a new `seek(provided_layer_index, provided_layer)` method to both Sealer and KeyGenerator instead of having two new `new_from_layer()` functions?

Reply:
This way we'll be able to reset the state of a Sealer/KeyGenerator without creating a new instance of them.